src/app/api/avclient.service.ts

```diff
@@ -1,136 +1,70 @@
 import { Injectable } from '@angular/core';
 
 import { StatuscodeService } from 'src/app/api/statuscode.service';
-import { Receipt } from 'src/app/class/receipt';
 import { VoterartifactsService } from 'src/app/api/voterartifacts.service';
+import { MockClient as MockClient } from './mockclient';
+import { IAVClient, AVClient, CastVoteRecord, BallotBoxReceipt, NistConverter } from '@aion-dk/js-client';
+import { environment } from '../../environments/environment';
 import { UserService } from 'src/app/class/user/user.service';
 
 @Injectable({
   providedIn: 'root',
 })
 export class AvclientService {
-  cachedAccessCode: any;
   serverURL: any;
+  client: IAVClient;
 
   constructor(
     public statuscodeService: StatuscodeService,
     public voterartifactsService: VoterartifactsService,
     private userService: UserService
   ) {}
 
+  // eslint-disable-next-line unused-imports/no-unused-vars
+  private static convertNIST103ToAvCvr(nistCvr: string): CastVoteRecord {
+    //console.log('Ready to parse NIST CVR', nistCvr);
+    return NistConverter.default.nistCvrToAvCvr(nistCvr);
+  }
+
   initServerURL(bulletinBoardURL) {
     this.serverURL = bulletinBoardURL; // to be used in other constructor/initializer calls
     if (this.userService.getUser() !== undefined) {
       this.voterartifactsService.initialize(this.userService.getUser().lastName);
     } // to be added: other initializer calls included the one deprecated below
-  }
 
-  initialize() {} // to be deprecated as an external interface
+    if (environment.production) {
+      this.client = new AVClient(bulletinBoardURL);
```

Comment:
I really like how this was done. This class does a good job of abstracting away the complexity of the individual calls to the AV client, while also separating out the mock functionality into its own class.